src/devices/blueline.c

```diff
@@ -0,0 +1,356 @@
+            // but they don't line up with what my official LCD display shows.
+            //
+            // I'm leaving the display raw while I figure this out better...
+            //
+            // A: deg_f = 0.823 * recvd_temp - 28.63
+            // B: deg_c = 0.457 * recvd_temp - 33.68
+
+            const int8_t temperature = offset_payload_u8[1];
+            const uint8_t flags = offset_payload_u8[0] >> 2;
+            const uint8_t battery = (flags & 0x20) >> 5;
+            /* clang-format off */
+            data = data_make(
+                    "model",         "", DATA_STRING, BLUELINE_MODEL,
+                    "id",            "", DATA_FORMAT, "%04x", DATA_INT, context->current_sensor_id,
+                    "flags",         "", DATA_FORMAT, "%02x", DATA_INT, flags,
+                    "battery_ok",    "", DATA_INT,    !battery,
+                    "temperature_C", "", DATA_INT,    temperature,
+                    "mic",           "", DATA_STRING, "CRC",
+                    NULL);
+            /* clang-format on */
+            decoder_output_data(decoder, data);
+            payloads_decoded++;
+        } else { // Assume BLUELINE_ENERGY_MSG
+            // (The lowest two bits of the pulse count will always be the same because message_type is overlaid there)
+            const uint16_t pulses = offset_payload_u16;
+            /* clang-format off */
+            data = data_make(
+                    "model",    "", DATA_STRING, BLUELINE_MODEL,
+                    "id",       "", DATA_FORMAT, "%04x", DATA_INT, context->current_sensor_id,
+                    "impulses", "", DATA_INT,    pulses,
+                    "mic",      "", DATA_STRING, "CRC",
+                    NULL);
+            /* clang-format on */
+            decoder_output_data(decoder, data);
+            payloads_decoded++;
+        }
+    }
+
+    return ((payloads_decoded > 0) ? payloads_decoded : most_applicable_failure);
+}
+
+static char *output_fields[] = {
+        "model",
+        "id",
+        "flags",
+        "gap",
+        "impulses",
+        "battery_ok",
+        "temperature_C",
+        "mic",
+        NULL,
+};
+
+r_device blueline;
+
+static r_device *blueline_create(char *arg)
+{
+    r_device *r_dev = create_device(&blueline);
+    if (!r_dev) {
+        fprintf(stderr, "blueline_create() failed\n");
+        return NULL; // NOTE: returns NULL on alloc failure.
+    }
+
+    struct blueline_context *context = malloc(sizeof(*context));
+    if (context == NULL) {
+        WARN_MALLOC("blueline_create()");
+        free(r_dev);
+        return NULL; // NOTE: returns NULL on alloc failure.
+    }
+    memset(context,0,sizeof(*context));
+    r_dev->decode_ctx = context;
+
+    if (arg != NULL) {
+        if (strcmp(arg, "auto") == 0) {
+            // Setup for auto identification
+            context->searching_for_new_id = true;
+            fprintf(stderr, "Blueline decoder will try to autodetect ID.\n");
```

Comment:
Verbose guard missing.

Reply:
I'm not seeing any way to determine if we're in verbose mode inside the creator functions.

Reply:
If you'd like, I can add a commit into this pull request to add more parameters to the create_fn, either the whole of the r_cfg_t or just the verbosity value from it.  I'll adjust fineoffset.c to match.

Reply:
Remove the output for now. Lets focus on getting this patch in first. Then we can add it later.

Reply:
Please don't access r_cfg from decoders -- we need some modularity.